### PR TITLE
[issue 140] Examples: demonstrate WildFly with Keycloack SAML adapter Galleon Pack s2i provisioning with Intersmash

### PR DESCRIPTION
**Pull request by fabiobrz**
## Description
An example that demonstrates provisioning WildFly on OpenShift via the s2i based image provisioner

Relates to #140 

## Type of change

 - [ ] Bug fix (non-breaking change which fixes an issue)
 - [ ] Feature (non-breaking change which adds functionality)
 - [ ] Breaking change (fix or feature that would cause existing functionality to change
 - [x] Chore (non-breaking change which doesn't affect codebase;
   test, version modification, documentation, etc.)

## Checklist
 - [x] Code contributed by me aligns with current project license: [Apache 2.0](https://www.apache.org/licenses/LICENSE-2.0)
 - [x] Pull Request contains a description of the changes
 - [x] Pull Request does not include fixes for multiple issues/topics
 - [x] Code is self-descriptive and/or documented
 - **N/A** I have implemented unit tests to cover my changes
 - [x] I tested my code in OpenShift

### CODE REVIEW

File: examples/wildfly-keycloak-saml-adapter/README.md [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+OpenShift, in order to generate the final immutable image containing the WildFly application server and the 
+application deployment itself.
+In our case the build input is represented by a local WildFly deployment, as generated by the WildFly Maven plugin 
+execution. 
+
+#### Resolving the build input
+The [templating-maven-plugin](https://www.mojohaus.org/templating-maven-plugin/) is used to have - for example - Maven 
+project properties injected with concrete values in Java classes at runtime. 
+
+When built, this example uses such plugin in order to generate the
```
Comment (tommaso-borgato) [inline examples/wildfly-keycloak-saml-adapter/README.md:37]: @fabiobrz is this related with "Resolving the build input"?

File: examples/wildfly-keycloak-saml-adapter/src/test/java/org/jboss/intersmash/examples/wildfly/keycloak/saml/KeycloakCapableImageBasedWildflyApplicationTest.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+
+import org.apache.http.HttpStatus;
+import org.jboss.intersmash.annotations.Intersmash;
+import org.jboss.intersmash.annotations.Service;
+import org.jboss.intersmash.annotations.ServiceUrl;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.Test;
+
+@Intersmash(@Service(KeycloakCapableImageBasedWildflyApplication.class))
+public class KeycloakCapableImageBasedWildflyApplicationTest {
```
Comment (tommaso-borgato) [inline examples/wildfly-keycloak-saml-adapter/src/test/java/org/jboss/intersmash/examples/wildfly/keycloak/saml/KeycloakCapableImageBasedWildflyApplicationTest.java:18]: @fabiobrz I don't understand how is this supposed to work when we don't have an actual Keycloak server....

**Comment by tommaso-borgato** [comment]
LGTM, merging

File: examples/wildfly-keycloak-saml-adapter/src/test/java/org/jboss/intersmash/examples/wildfly/keycloak/saml/KeycloakCapableImageBasedWildflyApplicationTest.java (hunk 2 above)
Comment (fabiobrz) [reply examples/wildfly-keycloak-saml-adapter/src/test/java/org/jboss/intersmash/examples/wildfly/keycloak/saml/KeycloakCapableImageBasedWildflyApplicationTest.java:18]: As discussed earlier, this is a WildFly service example, no Keycloak in place.
The only thing related to Keycloak is that the configuration of the Galleon-provisioned server includes the Keycloak SAML Adapter Galleon layer.
